project-base/storefront/components/Basic/ModalGallery/ModalGallery.tsx

```diff
@@ -0,0 +1,212 @@
+
+    const selectPreviousItem = () => {
+        setSelectedIndex(selectedIndex > 0 ? selectedIndex - 1 : lastItemIndex);
+    };
+
+    const selectNextItem = () => {
+        setSelectedIndex(selectedIndex < lastItemIndex ? selectedIndex + 1 : 0);
+    };
+
+    useEffect(() => {
+        setItemsRefs(
+            Array(items.length)
+                .fill(null)
+                .map(() => createRef()),
+        );
+    }, []);
+
+    useEffect(() => {
+        if (isCarouselDisplayed) {
+            itemsRefs?.[selectedIndex].current?.scrollIntoView({
+                behavior: 'smooth',
+                block: 'nearest',
+                inline: 'start',
+            });
+        }
+    }, [selectedIndex, itemsRefs, isCarouselDisplayed]);
+
+    useKeypress('Escape', onCloseModal);
+
+    useKeypress('ArrowRight', selectNextItem);
+
+    useKeypress('ArrowLeft', selectPreviousItem);
+
+    const handlers = useSwipeable({
+        onTap: onCloseModal,
+        onSwipedLeft: selectNextItem,
+        onSwipedRight: selectPreviousItem,
+        trackMouse: true,
+    });
+
+    return (
+        <div className="fixed inset-0 z-aboveOverlay flex select-none flex-col bg-dark p-2">
+            <div className="flex w-full flex-1 flex-col justify-center gap-2">
+                <div className="my-auto flex max-h-[80dvh] flex-1 items-center justify-center" {...handlers}>
+                    {isImage && (
+                        <Image
+                            key={selectedIndex}
+                            alt={selectedGalleryItem.name || `${galleryName}-${selectedIndex}`}
+                            className="max-h-full object-contain"
+                            draggable={false}
+                            height={800}
+                            sizes="(max-width: 768px) 100vw, 850px"
+                            src={selectedGalleryItem.url}
+                            width={800}
+                            onLoad={() => setIsLoaded(true)}
+                        />
+                    )}
+
+                    {isVideo && (
+                        <iframe
+                            allowFullScreen
+                            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+                            className="aspect-video max-h-full w-full max-w-xl md:max-w-[1500px]"
+                            src={`https://www.youtube.com/embed/${selectedGalleryItem.token}?autoplay=1&mute=1`}
+                            title={selectedGalleryItem.description}
+                        />
+                    )}
+                </div>
+
+                {isImage && selectedGalleryItem.name && (
+                    <div className="text-center text-greyDarker">{selectedGalleryItem.name}</div>
+                )}
+
+                <div className="flex items-center justify-center gap-2">
+                    {isCarouselDisplayed && (
```

Comment:
I suggest extract the Carousel into a separate component

Reply:
Yeah, I guess it will be better organized. Refactored.